bl1/bl1_main.c

```diff
@@ -44,14 +44,14 @@
  ******************************************************************************/
 static void __dead2 bl1_run_bl2(entry_point_info_t *bl2_ep)
 {
+	/* Check bl2 security state is expected as secure */
+	assert(GET_SECURITY_STATE(bl2_ep->h.attr) == SECURE);
+
 	bl1_arch_next_el_setup();
 
 	/* Tell next EL what we want done */
 	bl2_ep->args.arg0 = RUN_IMAGE;
 
-	if (GET_SECURITY_STATE(bl2_ep->h.attr) == NON_SECURE)
-		change_security_state(GET_SECURITY_STATE(bl2_ep->h.attr));
```

Comment:
I'm in my earlier comment [here](https://github.com/ARM-software/tf-issues/issues/314#issuecomment-124031548), I suggested that the `change_security_state` call should be left in, since future "firmware update" functionality may change the `SCR_NS_BIT`. However, after discussion with the "firmware update" author, we agreed that it's the responsibility of the "firmware update" code to ensure the `SCR_NS_BIT` is cleared by the time this code is reached.

However, this `bl1_run_bl2` function should have an additional assert that the `SCR_NS_BIT` is clear.